Content.Server/StationEvents/Events/GreytideVirusRule.cs

```diff
@@ -0,0 +1,96 @@
+using Content.Server.StationEvents.Components;
+using Content.Shared.Access;
+using Content.Shared.Access.Systems;
+using Content.Shared.Access.Components;
+using Content.Shared.Doors.Components;
+using Content.Shared.Doors.Systems;
+using Content.Shared.Lock;
+using Content.Shared.GameTicking.Components;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+
+namespace Content.Server.StationEvents.Events;
+
+
+/// <summary>
+///     Greytide Virus event
+///     This will open and bolt airlocks and unlock lockers from randomly selected access groups.
+/// </summary>
+public sealed class GreytideVirusRule : StationEventSystem<GreytideVirusRuleComponent>
+{
+    [Dependency] private readonly AccessReaderSystem _access = default!;
+    [Dependency] private readonly SharedDoorSystem _door = default!;
+    [Dependency] private readonly LockSystem _lock = default!;
+    [Dependency] private readonly IPrototypeManager _prototype = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    protected override void Added(EntityUid uid, GreytideVirusRuleComponent virusComp, GameRuleComponent gameRule, GameRuleAddedEvent args)
+    {
+        if (!TryComp<StationEventComponent>(uid, out var stationEvent))
+            return;
+
+        // pick severity randomly from range if not specified otherwise
+        virusComp.Severity ??= virusComp.SeverityRange.Next(_random);
+        virusComp.Severity = Math.Min(virusComp.Severity.Value, virusComp.AccessGroups.Count);
+
+        stationEvent.StartAnnouncement = Loc.GetString("station-event-greytide-virus-start-announcement", ("severity", virusComp.Severity.Value));
```

Comment:
we should be getting rid of such hardcoded strings

Reply:
Move it into the component maybe?

Reply:
I think it's fine as it's just the code-side of text that's customisable via localisation.